### ISSUE TITLE
General Issue: Consider use of the HEAD(default) branch as devel

### ISSUE THREAD

**Issue by galachad** [pharmaverse/admiral#1861]
### Background Information

In most of the case when user install package from GitHub it's install HEAD(default) branch.

This issue is to consider setting a HEAD branch to devel. 

Pros:
- More easy start for currently development version.
- For development it can simplify the start process
- Easy to put integration with codespace as devel will be default branch
- Less difficult to apply workflow changes (do not need to touch main branch)

Cons:
- It can impact CI
- Needs to verify how `{stage.dependency}` handle default branch.
- Can cause incompatibility when non experience user install from GitHub on prd environment

I do not find any topic about that, but maybe it was already discussed in the past.

**Comment by bms63** [comment]
@cicdguy and @dgrassellyb  can we get your expert opinions on this topic please

**Comment by bms63** [comment]
Is that not what we are doing? I googled a few pages for GitHub Flow and we pretty much follow that.  Also, we removed the `pre-release` branch so one less headache.

**Comment by galachad** [comment]
I agree with @cicdguy, but I haven't been thinking about modifying the branch strategy. In any case of using git flow or github flow having default branch as more recently development branch can simplify some process.

**Comment by bundfussr** [comment]
I think changing the default branch to `devel` makes sense because most users do not install admiral from github but from CRAN or other sources. So installing from github is mainly done by developers.
